app/models/project_metadata.rb

```diff
@@ -1,125 +1,120 @@
-  end
-
-  def generate_approval_events(note)
-    # create two provenance events, one for approving the project and another for changing the status of the project
-    project.provenance_events.create(event_type: ProvenanceEvent::APPROVAL_EVENT_TYPE, event_person: current_user.uid, event_details: "Approved by #{current_user.display_name_safe}",
-                                     event_note: note)
-    project.provenance_events.create(event_type: ProvenanceEvent::STATUS_UPDATE_EVENT_TYPE, event_person: current_user.uid, event_details: "The Status of this project has been set to approved")
+  # rubocop:disable Metrics/MethodLength
+  def initialize_from_hash(metadata_hash)
+    @title = metadata_hash[:title]
+    @description = metadata_hash[:description]
+    @status = metadata_hash[:status]
+    @data_sponsor = metadata_hash[:data_sponsor]
+    @data_manager = metadata_hash[:data_manager]
+    @departments = metadata_hash[:departments]
+    @ro_users = metadata_hash[:ro_users]
+    @rw_users = metadata_hash[:rw_users]
+
+    @project_id = metadata_hash[:project_id]
+    @project_purpose = metadata_hash[:project_purpose]
+    @project_directory = metadata_hash[:project_directory]
+
+    @storage_capacity = metadata_hash[:storage_capacity]
+    @storage_performance_expectations = metadata_hash[:storage_performance_expectations]
+    @created_on = metadata_hash[:created_on]
+    @created_by = metadata_hash[:created_by]
+    @updated_by = metadata_hash[:updated_by]
+    @updated_on = metadata_hash[:updated_on]
+    set_defaults
   end
+  # rubocop:enable Metrics/MethodLength
 
-  def create(params:)
-    project.metadata = update_metadata(params: params)
-    if project.valid? && project.metadata["project_id"].blank?
-      draft_doi
-    end
-    project.metadata["project_id"]
+  # Initializes the object with the values in the params (which is an ActionController::Parameters)
+  # rubocop:disable Metrics/MethodLength
+  def initialize_from_params(params)
+    @title = params[:title]
+    @description = params[:description]
+    @status = params[:status] if params[:status]
+    @data_sponsor = params[:data_sponsor]
+    @data_manager = params[:data_manager]
+    @departments = params[:departments]
+    @ro_users = user_list_params(params, read_only_counter(params), "ro_user_")
+    @rw_users = user_list_params(params, read_write_counter(params), "rw_user_")
+
+    @project_id = params[:project_id]
+    @project_purpose = params[:project_purpose]
+    @project_directory = params[:project_directory]
+
+    @storage_capacity = params[:storage_capacity]
+    @storage_performance_expectations = params[:storage_performance_expectations]
+
+    @created_by = params[:created_by] if params[:created_by]
+    @created_on = params[:created_on] if params[:created_on]
+    @updated_by = params[:updated_by] if params[:updated_by]
+    @updated_on = params[:updated_on] if params[:updated_on]
+    set_defaults
```

Comment:
Using this version of the defaults will cause a regression. We need to be utilizing the Rails.configuration.project_defaults here

Reply:
Good catch, we are not reading them from the config again